src/mlwm/build_inference_artifact.py

```diff
@@ -0,0 +1,350 @@
+import sys
+import tempfile
+import zipfile
+from pathlib import Path
+
+import dotenv
+import mllam_data_prep as mdp
+import xarray as xr
+import yaml
+from loguru import logger
+from tqdm import tqdm
+from upath import UPath
+
+from . import __version__
+
+DEFAULT_ARTIFACT_PATH_FORMAT = (
+    "s3://mlwm-artifacts/inference-artifacts/{artifact_name}.zip"
+)
+
+dotenv.load_dotenv()
+
+ARTIFACT_PATH_FORMAT = os.environ.get(
+    "ARTIFACT_PATH_FORMAT", DEFAULT_ARTIFACT_PATH_FORMAT
+)
+
+TRAINING_CLI_ARGS_FILE = "training_cli_args.yaml"
+
+
+def _find_datastore_paths(nl_config_path: str):
+    with open(nl_config_path, "r") as f:
+        nl_config = yaml.safe_load(f)
+
+    def _make_abspath(datastore_path: str):
+        """
+        Make the path absolute if it is not already. If the path is relative,
+        it is made absolute with respect to the neural-lam config file.
+        """
+        # if the path is absolute, return it as is
+        if Path(datastore_path).is_absolute():
+            return datastore_path
+        return Path(nl_config_path).parent / datastore_path
+
+    datastore_paths = {}
+    if "datastore" in nl_config:
+        datastore_paths[None] = _make_abspath(
+            nl_config["datastore"]["config_path"]
+        )
+    elif "datastores" in nl_config:
+        for name, datastore_config in nl_config["datastores"].items():
+            datastore_paths[name] = _make_abspath(
+                datastore_config["config_path"]
+            )
+    else:
+        raise ValueError("No datastore found in the config file.")
+
+    return datastore_paths
+
+
+def _extract_stats(datastore_config_path: str, artifact_path: str):
+    """
+    Extract statistics from the training dataset created from the datastore
+    configuration file. The statistics are saved to a file named
+    {datastore_name}.stats.zarr in the artifact output directory.
+
+    Parameters
+    ----------
+    datastore_config_path : str
+        Path to the datastore configuration file. The name of the
+        datastore is assumed to be the name of the file without the extension, i.e.
+        {datastore_name}.yaml.
+    artifact_path : str
+        Path to the artifact output directory. The statistics will be saved
+        to a file named {datastore_name}.stats.zarr in this directory.
+    """
+    fp_stats = (
+        Path(artifact_path)
+        / Path(datastore_config_path).with_suffix(".stats.zarr").name
+    )
+
+    dataset_path = Path(datastore_config_path).with_suffix(".zarr")
+    ds = xr.open_zarr(dataset_path)
+
+    ds_stats = xr.Dataset()
+
+    # XXX: this is a massive hack, but for now we will just assume that any
+    # variable that has "__" in its name twice is a statistic. This could be
+    # done better by using the implementation in mllam-data-prep...
+
+    for var_name in ds.data_vars:
+        if len(var_name.split("__")) == 3:
+            # this is a statistic
+            ds_stats[var_name] = ds[var_name].compute()
```

Comment:
In the mdp datastore, is statistical variable named with two _ ? 
I think I need a note on why "__" is necessary.

Reply:
Good point. It is because that is the variable name format I have picked 😇 : https://github.com/mllam/mllam-data-prep/blob/v0.6.0/mllam_data_prep/create_dataset.py#L279. I will add a comment referencing this

Reply:
done in https://github.com/leifdenby/mlwm-deployment/commit/36a5d20915257aae6678dff8c2803bde7149c679